bot/exts/valentines/valentine_zodiac.py

```diff
@@ -19,36 +22,115 @@ class ValentineZodiac(commands.Cog):
 
     def __init__(self, bot: commands.Bot):
         self.bot = bot
-        self.zodiacs = self.load_json()
+        self.zodiacs, self.zodiac_fact = self.load_comp_json()
 
     @staticmethod
-    def load_json() -> dict:
+    def load_comp_json() -> Tuple[dict, dict]:
         """Load zodiac compatibility from static JSON resource."""
-        p = Path("bot/resources/valentines/zodiac_compatibility.json")
-        with p.open(encoding="utf8") as json_data:
+        explanation_file = Path("bot/resources/valentines/zodiac_explanation.json")
+        compatibility_file = Path("bot/resources/valentines/zodiac_compatibility.json")
+        with explanation_file.open(encoding="utf8") as json_data:
+            zodiac_fact = load(json_data)
+            for zodiac_data in zodiac_fact.values():
+                zodiac_data['start_at'] = date.fromisoformat(zodiac_data['start_at'])
+                zodiac_data['end_at'] = date.fromisoformat(zodiac_data['end_at'])
+        with compatibility_file.open(encoding="utf8") as json_data:
             zodiacs = load(json_data)
-            return zodiacs
+        return zodiacs, zodiac_fact
 
-    @commands.command(name="partnerzodiac")
-    async def counter_zodiac(self, ctx: commands.Context, zodiac_sign: str) -> None:
+    def error(self, zodiac: str) -> discord.Embed:
+        """Returns error embed."""
+        embed = discord.Embed()
+        embed.color = Colours.pink
+        err_comp = [f"`{i}` {zod_name}" for i, zod_name in enumerate(self.zodiac_fact.keys(), start=1)]
+        err = ("\n").join(err_comp)
+        error_msg = f"**{zodiac}** is not a valid zodiac sign, here is the list of valid zodiac signs."
+        embed.description = f"{error_msg}\n{err}"
```

Comment:
a lot of un-necessary variables here.
you can replace with this
```suggestion
        embed.description = f"**{zodiac}** is not a valid zodiac sign, here is the list of valid zodiac signs.\n"
        for i, zod_name in enumerate(self.zodiac_fact.keys(), start=1):
            embed.description += f"`{i}` {zod_name}\n"
```